src/plugins/ImportKeras/ImportKeras.js

```diff
@@ -0,0 +1,307 @@
+                        }
+                    }
+                }
+            });
+        }
+    };
+
+    // This method is used to convert javascript arrays to a tuple/ list in string Representation.
+    ImportKeras.prototype._Stringify = function (obj) {
+        if (obj instanceof Array) {
+            return '[' + obj.map((val) => {
+                if (val) {
+                    return val;
+                } else {
+                    return "None";
+                }
+            }).join(', ') + ']';
+        } else {
+            return obj;
+        }
+    };
+
+    // This method is used to convert various classes from the keras JSON to deepforge meta Nodes
+    ImportKeras.prototype._mapClassName = function (orgName) {
+        let classMap = ModelMaps.CLASS_MAP;
+        if (Object.keys(classMap).indexOf(orgName) > -1) {
+            return classMap[orgName];
+        } else {
+            return orgName;
+        }
+    };
+
+    ImportKeras.prototype._mapArgName = function (orgName) {
+        let argMap = ModelMaps.ARGUMENTS_MAP;
+        if (Object.keys(argMap).indexOf(orgName) > -1) {
+            return argMap[orgName];
+        } else {
+            return orgName;
+        }
+    };
+    /**********************The functions below Add Connections between the Layers**************/
+    ImportKeras.prototype.addConnections = function () {
+        // this._findNodeByName();
+        let layers = this.modelInfo.layers;
+        let layerInputConnections = {};
+        let layerOutputConnections = {};
+        let connections = null;
+        layers.forEach((layer) => {
+            layerInputConnections[layer.name] = [];
+            layerOutputConnections[layer.name] = [];
+        });
+
+        layers.forEach((layer) => {
+            if (layer.inbound_nodes.length > 0) {
+                connections = layer.inbound_nodes;
+                connections.forEach((connection) => {
+
+                        if (this._layerNameExists(connection)) {
+                            layerInputConnections[layer.name].push(connection);
+                            layerOutputConnections[connection].push(layer.name);
+                    }
+                });
+
+            }
+        });
+
+        return this._updateConnections(layerInputConnections, layerOutputConnections);
+
+    };
+
+
+    ImportKeras.prototype._layerNameExists = function (layerName) {
+        let allLayerNames = this.modelInfo.layers.map((layer) => {
+            return layer.name;
+        });
+
+        return allLayerNames.indexOf(layerName) > -1;
+    };
+
+    ImportKeras.prototype._updateConnections = function (inputs, outputs) {
+        let allLayerNames = Object.keys(inputs);
+        return Promise.all(allLayerNames.map((layerName) => {
+            let dstLayer = this.layerInfo[layerName];
+            let srcs = inputs[layerName];
+            if (srcs.length > 0) {
+                return Promise.all(srcs.map((src, index, records) => {

```

Comment:
Once you change `_connectLayers` so it isn't async, you can remove `Promise.all`